vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc

```diff
@@ -135,24 +156,36 @@ void LedgerImpl::Initialize(
   }
 
   initializing_ = true;
-
   InitializeDatabase(execute_create_script, callback);
 }
 
 void LedgerImpl::InitializeDatabase(
     const bool execute_create_script,
     ledger::ResultCallback callback) {
-  ledger::ResultCallback finish_callback =
-      std::bind(&LedgerImpl::OnInitialized,
-          this,
-          _1,
-          std::move(callback));
+  ledger::ResultCallback finish_callback = std::bind(
+      &LedgerImpl::OnInitialized,
+      this,
+      _1,
+      std::move(callback));
 
   auto database_callback = std::bind(&LedgerImpl::OnDatabaseInitialized,
       this,
       _1,
       finish_callback);
-  bat_database_->Initialize(execute_create_script, database_callback);
+  database()->Initialize(execute_create_script, database_callback);
+}
+
+void LedgerImpl::OnInitialized(
+    const ledger::Result result,
+    ledger::ResultCallback callback) {
+  initializing_ = false;
+  callback(result);
```

Comment:
So it's OK to call the initialized callback before calling `StartServices` below? Is initialization a separate concept?

Reply:
this PR doesn't change the logic, but I agree that would be good to be called at the end

Reply:
fixed